apps/prairielearn/src/pages/elementFiles/elementFiles.sql

```diff
@@ -0,0 +1,22 @@
+-- BLOCK select_has_shared_question
+SELECT
+  EXISTS (
+    SELECT
+      1
+    FROM
+      questions AS q
+    WHERE
+      q.shared_publicly
+      AND course_id = $producing_course_id
+    UNION
+    SELECT
+      1
+    FROM
+      sharing_sets AS ss
+      JOIN sharing_set_courses AS ssc ON ss.id = ssc.sharing_set_id
+      JOIN sharing_set_questions AS ssq ON ss.id = ssq.sharing_set_id
+      JOIN questions AS q on q.id = ssq.question_id
```

Comment:
Do we want to actually check if there is a question in the sharing set here? or get rid of these two lines and just check if the sharing set exists?

Reply:
It should be cheap enough to check for the existence of an actual shared question; I'm fine to leave this as is.